backend/src/main/java/atlantbh/restaurants/services/BaseService.java

```diff
@@ -0,0 +1,44 @@
+package atlantbh.restaurants.services;
+
+import atlantbh.restaurants.models.BaseModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public class BaseService<M extends BaseModel, R extends JpaRepository<M,Long>> {
+
+    @Autowired
+    protected R repository;
+
+    public Optional<M> getById(Long id) {
+        return repository.findById(id);
+    }
+
+    public void save(M m) {
+        repository.save(m);
+    }
+
+    public Collection<M> all() {
+        return repository.findAll();
+    }
+
+    public Long count() {
+        return repository.count();
+    }
+
+    public void delete(M m) {
+        repository.delete(m);
+    }
+
+    public void deleteById(Long id) { repository.deleteById(id) ;}
+
+    public void deleteAll() {
```

Comment:
You should avoid method like this.. Imagine you have 9384892340 data in db table, and you call delete all.. It is high load, or imagine one of them fails.. transaction will fail also.. Always stick to one request/one response meaning that if you select 100 checkboxes from frontend, it will be 100 promises, where you will have 100 `delete/:id` requests..

Reply:
Fixed c2ed9b4060983e1312e68ee9cdd1de2be1986207